File: DeviantArtFs.Types/StashMetadata.cs
using System;
using System.Collections.Generic;

namespace DeviantArtFs.Types {
    public class StashSubmission {
        public string? file_size { get; set; }
        public string? resolution { get; set; }
        public DeviantArtSubmittedWith? submitted_with { get; set; }
    }

    public class StashStats {
        public int? views { get; set; }
        public int? views_today { get; set; }
        public int? downloads { get; set; }
        public int? downloads_today { get; set; }
    }

    public class StashMetadata {
        public string title { get; set; } = "";
        public string? path { get; set; }
        public int? size { get; set; }
        public string? description { get; set; }
        public long? parentid { get; set; }
        public DeviationPreview thumb { get; set; } = new DeviationPreview();
        public string? artist_comments { get; set; }
        public string? original_url { get; set; }
        public string? category { get; set; }
        public DateTimeOffset? creation_time { get; set; }
        public DeviationPreview[]? files { get; set; }
        public StashSubmission? submission { get; set; }
        public StashStats? stats { get; set; }
        public Dictionary<string, string>? camera { get; set; }
        public long? stackid { get; set; }
        public long? itemid { get; set; }
        public string[]? tags { get; set; }
    }
}

File: DeviantArtFs/ResponseTypes/StashMetadata.fs
namespace DeviantArtFs.ResponseTypes

open System
open FSharp.Json

type StashSubmission = {
    file_size: string option
    resolution: string option
    submitted_with: SubmittedWith option
}

type StashStats = {
    views: int option
    views_today: int option
    downloads: int option
    downloads_today: int option
}

type StashMetadata = {
    title: string
    path: string option
    size: int option
    description: string option
    parentid: int64 option
    thumb: Preview option
    artist_comments: string option
    original_url: string option
    category: string option
    [<JsonField(Transform=typeof<Transforms.DateTimeOffsetEpoch>)>]
    creation_time: DateTimeOffset option
    files: Preview list option
    submission: StashSubmission option
    stats: StashStats option
    camera: Map<string, string> option
    stackid: int64 option
    itemid: int64 option
    tags: string list option
}